### PR TITLE
Issue #SB-25033 feat: Enable collaboration functionality for courses in the workspace

### PR DESCRIPTION
**Pull request by rajnishdargan**

### CODE REVIEW

File: projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+import * as _ from 'lodash-es';
+@Component({
+  selector: 'lib-manage-collaborator',
+  templateUrl: './manage-collaborator.component.html',
+  styleUrls: ['./manage-collaborator.component.scss']
+})
+export class ManageCollaboratorComponent implements OnInit {
+  @Output() modalDismissEmitter = new EventEmitter<any>();
+  @ViewChild('modal', { static: false }) private modal;
```
Comment (venkateshwarans) [inline projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts:15]: `static: false` is remove in angular 9, right?

File: projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            this.searchRes.content = [];
+          }
+        }
+    },
+    (error) => {
+      this.toasterService.error(_.get(this.configService, 'labelConfig.messages.error.001'));
+      console.log(error);
+      this.searchRes.content = [];
+      this.searchRes.isEmptyResponse = true;
+      this.searchRes.errorMessage = 'Oops! Something went wrong. Please try again later.';
```
Comment (venkateshwarans) [inline projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts:238]: Why can't we use the same error message as  
```suggestion
      this.searchRes.errorMessage = _.get(this.configService, 'labelConfig.messages.error.001')
```

File: projects/collection-editor-library/src/lib/components/editor/editor.component.html [hunk 3]
```diff
@@ -105,6 +105,9 @@
     </lib-question>
   </ng-container>
 </ng-container>
+<ng-container>
+  <lib-manage-collaborator *ngIf="addCollaborator" [addCollaborator]="addCollaborator" [collectionId]="collectionId" (modalDismissEmitter)="handleModalDismiss($event)"></lib-manage-collaborator>
```
Comment (venkateshwarans) [inline projects/collection-editor-library/src/lib/components/editor/editor.component.html:109]:
```suggestion
<ng-container *ngIf="addCollaborator">
  <lib-manage-collaborator [addCollaborator]="addCollaborator" [collectionId]="collectionId" (modalDismissEmitter)="handleModalDismiss($event)"></lib-manage-collaborator>
```

File: projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    ) { }
+
+  ngOnInit() {
+    this.currentUser = _.get(this.editorService.editorConfig, 'context.user');
+    this.isRootOrgAdmin = _.has(this.currentUser, 'isRootOrgAdmin') ? this.currentUser.isRootOrgAdmin : false;
+    this.userSearchBody.request.filters.rootOrgId = this.currentUser.orgIds;
+    this.getCollbaoratorAndCreator();
+  }
+
+  getCollbaoratorAndCreator() {
```
Comment (venkateshwarans) [inline projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts:61]: `getCollaboratorAndCreator` spelling.

File: projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+      console.log(error);
+    });
+  }
+
+  excludeCreatorAndCollaborators(allUsers) {
+    const users = _.filter(allUsers, user => {
+      return !_.includes(this.creatorAndCollaboratorsIds, user.identifier);
+    });
+    // return users; // line to be kept
+    return allUsers; // line to be removed
```
Comment (venkateshwarans) [inline projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts:114]: which one to be kept?

**Review by venkateshwarans — COMMENTED** [review]
check comments

File: projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts (hunk 1 above)
Comment (rajnishdargan) [reply projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts]: removed static: false

File: projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts (hunk 4 above)
Comment (rajnishdargan) [reply projects/collection-editor-library/src/lib/components/manage-collaborator/manage-collaborator.component.ts]: corrected

File: projects/collection-editor-library/src/lib/components/editor/editor.component.html (hunk 3 above)
Comment (rajnishdargan) [reply projects/collection-editor-library/src/lib/components/editor/editor.component.html]: done

File: projects/collection-editor-library/src/lib/services/helper/helper.service.ts [hunk 6]
```diff
@@ -100,6 +100,28 @@ export class HelperService {
...
+        request: userSearchBody.request
+      }
+    };
 
+    return this.publicDataService.post(req);
+  }
 
+  updateCollaborator(contentId, collaboratorList) {
+    const req = {
+      url: `content/v1/collaborator/update/` + contentId,
```
Comment (pallakartheekreddy) [inline projects/collection-editor-library/src/lib/services/helper/helper.service.ts:116]: @rajnishdargan please take URL from `url.config.json`

File: projects/collection-editor-library/src/lib/services/helper/helper.service.ts [hunk 7]
```diff
@@ -100,6 +100,28 @@ export class HelperService {
     }
   }
 
+  getAllUser(userSearchBody) {
+    const req = {
+      url: `user/v1/search?fields=orgName`,
```
Comment (pallakartheekreddy) [inline projects/collection-editor-library/src/lib/services/helper/helper.service.ts:105]: @rajnishdargan please take URL from url.config.json

**Review by pallakartheekreddy — CHANGES_REQUESTED** [review]

File: projects/collection-editor-library/src/lib/services/helper/helper.service.ts (hunk 6 above)
Comment (rajnishdargan) [reply projects/collection-editor-library/src/lib/services/helper/helper.service.ts]: @pallakartheekreddy Done, please verify

File: projects/collection-editor-library/src/lib/services/helper/helper.service.ts (hunk 7 above)
Comment (rajnishdargan) [reply projects/collection-editor-library/src/lib/services/helper/helper.service.ts]: @pallakartheekreddy Done, please verify.

**Review by pallakartheekreddy — APPROVED** [review]

**Review by pallakartheekreddy — APPROVED** [review]

**Pull request merged**